app/components/AlertDialog/__tests__/AlertDialog.test.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import AlertDialog from '..';
+
+jest.mock('@/services/api');
+
+describe('AlertDialog', () => {
+  let component = render(
+    <AlertDialog
+      isOpen={true}
+      onClose={() => { }}
+      title='Dialog Title'
+      text='Mock dialog text'
+    />
+  );
+
+  test('It renders', () => {
+    const { getByText, asFragment } = component;
+    expect(asFragment()).toMatchSnapshot();
+    expect(getByText('Dialog Title')).toBeInTheDocument()
+    expect(getByText('Mock dialog text')).toBeInTheDocument()
+  });
+
+  test('It allows custom confirm and cancel text', () => {
+    component = render(
+      <AlertDialog
+        isOpen={true}
+        onClose={() => { }}
+        title='Dialog Title'
+        text='Mock dialog text'
+        confirmText='yas'
+        cancelText='nyet'
+      />
+    );
+    const { getByText } = component;
+    expect(getByText('yas')).toBeInTheDocument()
+    expect(getByText('nyet')).toBeInTheDocument()
+  })
+
+
+  test('It calls onClose with the correct params', () => {
```

Comment:
I would split this into 2 tests, one for returns true for confirmed when confirm is clicked and one for returns false for confirmed when cancel is clicked